File: 53.最大子数组和.cs
/*
 * @lc app=leetcode.cn id=53 lang=csharp
 *
 * [53] 最大子数组和
 */

// @lc code=start
public partial class Solution
{
    public int MaxSubArray(int[] nums)
    {
        int len = nums.Length;
        if (len == 0)
        {
            return 0;
        }
        int max = int.MinValue;
        int sum = int.MinValue;
        for (int i = 0; i < len; i++)
        {
            int num = nums[i];
            if (sum < 0)
            {
                sum = num;
            }
            else
            {
                sum = sum + num;
            }
            if (sum > max)
            {
                max = sum;
            }
        }
        return max;
    }
}
// @lc code=end

File: src/53.最大子数组和.rs

/*
 * @lc app=leetcode.cn id=53 lang=rust
 *
 * [53] 最大子数组和
 */

// @lc code=start
impl Solution {
    pub fn max_sub_array(nums: Vec<i32>) -> i32 {
        let mut max = i32::MIN;
        let mut sum = i32::MIN;
        for num in nums {
            if sum < 0 {
                sum = num;
            } else {
                sum += num;
            }
            if sum > max {
                max = sum;
            }
        }
        return max;
    }
}
// @lc code=end
